### DISCUSSION TITLE
Where to find guide for integrating kratos with next js 14 and later

### DISCUSSION THREAD

**Discussion by chil-guy** [ory/kratos#4227] [Q&A] [answered]
Hello all, can anyone please share any guide, demo, template, or blog that i can follow to integrate next js 14 or later i.e. app router. i am trying to do that same since a month but unable to create good architecture. Thanks

**Comment by MarkusThielker** [comment]
I would recommend to have a look at the [official Next.js example](https://github.com/ory/kratos-selfservice-ui-react-nextjs). It does use the pages router but the logic and implementation of rendering the flows dynamically is the same. You can find that implementation in the 'pkg' directory.

**Reply by chil-guy** [reply]
Thanks for replay @MarkusThielker,
In the end i used barebone APIs for the browser workflow, as we had only few apis to work on this project. The official next js example uses a proxy([api/.ory](https://github.com/ory/kratos-selfservice-ui-react-nextjs/tree/master/pages/api/.ory)) which was not working with app router. Anyway thanks for the heads up. 
And yes the pkg folder did helped in defining the structure of the API objects and UI. Thanks again

**Comment by vinckr** [comment, marked as answer]
See here for an example with Ory Elements and app router @chil-guy @MarkusThielker https://github.com/ory/elements/tree/main/examples/nextjs-app-router

its still being worked on but very close to official release. 
fyi @jonas-jonas

**Reply by chil-guy** [reply]
Thanks for the reply
